examples/php/route_guide/RouteGuideService.php

```diff
@@ -0,0 +1,182 @@
+ *
+ */
+
+
+class RouteGuideService extends \Routeguide\RouteGuideServiceStub
+{
+    public function __construct($dbFilePath)
+    {
+        $dbFilePath = $dbFilePath ?? dirname(__FILE__) . '/route_guide_db.json';
+        $dbData = file_get_contents($dbFilePath);
+        if (!$dbData) {
+            throw new InvalidArgumentException(
+                "Error reading route db file: " . $dbFilePath
+            );
+        }
+        $featureList = json_decode($dbData);
+        if (!$featureList) {
+            throw new InvalidArgumentException(
+                "Error decoding route db file: " . $dbFilePath
+            );
+        }
+        foreach ($featureList as $feature) {
+            array_push($this->featureList, new Routeguide\Feature([
+                'name' => $feature->name,
+                'location' => new Routeguide\Point([
+                    'latitude' => $feature->location->latitude,
+                    'longitude' => $feature->location->longitude,
+                ]),
+            ]));
+        }
+    }
+
+    private function findFeature(\Routeguide\Point $point)
+    {
+        foreach ($this->featureList as $feature) {
+            $location = $feature->getLocation();
+            if (
+                $location->getLatitude() === $point->getLatitude()
+                && $location->getLongitude() === $point->getLongitude()
+            ) {
+                return $feature;
+            }
+        }
+        return null;
+    }
+
+    // The formula is based on http://mathforum.org/library/drmath/view/51879.html
+    private function calculateDistance(
+        \Routeguide\Point $start,
+        \Routeguide\Point $end
+    ) {
+        $toRadians = function (float $num) {
+            return $num * 3.1415926 / 180;
+        };
+        $coordFactor = 10000000.0;
+        $R = 6371000; // metres
+
+        $lat_1 = $start->getLatitude() / $coordFactor;
+        $lat_2 = $end->getLatitude() / $coordFactor;
+        $lon_1 = $start->getLongitude() / $coordFactor;
+        $lon_2 = $end->getLongitude() / $coordFactor;
+        $lat_rad_1 = $toRadians($lat_1);
+        $lat_rad_2 = $toRadians($lat_2);
+        $delta_lat_rad = $toRadians($lat_2 - $lat_1);
+        $delta_lon_rad = $toRadians($lon_2 - $lon_1);
+
+        $a = pow(sin($delta_lat_rad / 2), 2) +
+            cos($lat_rad_1) * cos($lat_rad_2) * pow(sin($delta_lon_rad / 2), 2);
+        $c = 2 * atan2(sqrt($a), sqrt(1 - $a));
+
+        return $R * $c;
+    }
+
+    public function GetFeature(
+        \Routeguide\Point $request,
+        \Grpc\ServerContext $serverContext
+    ): array {
+        $feature = $this->findFeature($request);
+        $notFoundFeature = new Routeguide\Feature([
+            'name' => '',
+            'location' => $request,
+        ]);
+        return [$feature ?? $notFoundFeature];
+    }
+
+    public function ListFeatures(
+        \Routeguide\Rectangle $request,
+        \Grpc\ServerCallWriter $writter,
```

Comment:
Typo, `$writer`. It's also in the code generator and generated code too.

Reply:
fixed